jenkins/check_projects.py

```diff
@@ -39,190 +41,393 @@
 import subprocess
 import sys
 import threading
+import logging
 
-semaphore = threading.Semaphore(10)  # 10 concurrent gcloud calls
-lock = threading.Lock()  # Don't print/append concurrently
-null = open('/dev/null', 'w')
-
-# Not thread safe, acquire lock before accessing.
-completed = set()  # Projects with no updated needed
-updated = set()  # Projects which changed
-errors = set()  # Projects which failed to fully update
-helpers = {}  # People with owners rights to update error projects
+logger = logging.getLogger('check_project')
 
 DEFAULT = {
-    'roles/editor': [
+    'IAM': {
+      'roles/editor': [
         'serviceAccount:kubekins@kubernetes-jenkins.iam.gserviceaccount.com',
-        'serviceAccount:pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com',
-    ],
+        'serviceAccount:pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com'],
+    },
+    'EnableVMZonalDNS': False,
 }
 
 
+class RateLimitedExec(object):
+  def __init__(self):
+    self._semaphore = threading.Semaphore(10)  # 10 concurrent gcloud calls
+
+  def check_output(self, *args, **kwargs):
+    with self._semaphore:
+      return subprocess.check_output(*args, **kwargs)
+
+  def call(self, *args, **kwargs):
+    with self._semaphore:
+      return subprocess.call(*args, **kwargs)
+
+
+class Results(object):
+  class Info(object):
+    def __init__(self):
+      self.updated = False
+      self.errors = [] # errors is list of strings describing the error context
+      self.helpers = []  # People with owners rights to update error projects
+
+  def __init__(self):
+    self._lock = threading.Lock()
+    self._projects = {} # project -> Info
+
+  @property
+  def errors(self):
+    """
+    |return| set of projects that have errors.
+    """
+    return set([key for key in self._projects if self._projects[key].errors])
+
+  @property
+  def projects(self):
+    return self._projects
+
+  def reportProject(self, project):
```

Comment:
Expected style is `function_name` not `functionName`

https://google.github.io/styleguide/pyguide.html?showone=Naming#Naming